docs/root/configuration/http_filters/router_filter.rst

```diff
@@ -80,6 +80,10 @@ using a ',' delimited list. The supported policies are:
     :ref:`config_http_filters_router_x-envoy-upstream-rq-timeout-ms` is an outer time limit for a
     request, including any retries that take place.
 
+gateway-error
+  Envoy will attempt a retry if the upstream server responds with 502, 503, or 504 response code.
```

Comment:
I would recommend deleting this line. It's similar including the text about disconnect/reset/read-timeout, etc. (This is related to my other comment in the implementation PR). *Unless* of course you intend it really to be only upstream responses. I assume though you want Envoy origin responses also?